lib/backend_common/backend_common/auth0.py

```diff
@@ -76,7 +76,7 @@ def decorated(*args, **kwargs):
             response = requests.get(url, params=payload)
 
             # Because auth0 returns http 200 even if the token is invalid.
-            if response.content == b'Unauthorized':
+            if response.content == b'Unauthorized' or not response.ok:
```

Comment:
Maybe it would be better to even turn that if statement around and skip checking the `response.content` then.
I will +1 the PR but maybe you find below code also more readable.


```python
if response.ok:
    # store response.content for later
    flask.g.userinfo = json.loads(str(response.content, 'utf-8'))
    # g.oidc_id_token = token # requires a specific format
    flask.g.access_token = token
    return view_func(*args, **kwargs)

response_body = {
    'error': 'invalid_token',
    'error_description': str(response.content, 'utf-8'),
}
if render_errors:
    response_body = json.dumps(response_body)
return response_body, 401, {'WWW-Authenticate': 'Bearer'}
```

Reply:
Yeah, definitely this is more readable. Let me confirm the status of `response.content`, then I'll update the PR.

Reply:
@rail ping

Reply:
I've been keeping an eye on this at the same time with exploring auth0 features. I definitely can merge the initial patch, but if there is no rush I'd like to make sure that we don't break anything with more refactoring. I bet there was a good reason for the initial implementation, where we don't trust 200.